src/materials/MeshRefractionMaterial.tsx

```diff
@@ -130,10 +124,15 @@ export const MeshRefractionMaterial = shaderMaterial(
   #endif
   
   void main() {
-    mat4 modelMatrixInverse = inverse(modelMatrix * vInstanceMatrix);
+    #ifdef USE_INSTANCING
+      mat4 modelMatrixInverse = inverse(modelMatrix * vInstanceMatrix);
+    #else
+      mat4 modelMatrixInverse = inverse(modelMatrix);
+    #endif
+
     vec2 uv = gl_FragCoord.xy / resolution;
-    vec3 directionCamPerfect = (projectionMatrixInv * vec4(uv * 2.0 - 1.0, 0.0, 1.0)).xyz;
-    directionCamPerfect = (viewMatrixInv * vec4(directionCamPerfect, 0.0)).xyz;
+    vec3 directionCamPerfect = (inverse(projectionMatrix) * vec4(uv * 2.0 - 1.0, 0.0, 1.0)).xyz;
+    directionCamPerfect = (inverse(viewMatrix) * vec4(directionCamPerfect, 0.0)).xyz;
```

Comment:
`inverse` in a fragment shader should be avoided.  I believe these can be calculated in the [react component](https://github.com/pmndrs/drei/blob/master/src/core/MeshRefractionMaterial.tsx) using a beforeRender callback like [MeshReflectorMaterial](https://github.com/pmndrs/drei/blob/master/src/core/MeshReflectorMaterial.tsx).  the tradeoff is that people using the pure threejs version of the material will need to compute it themselves, but that's already true for a number of the other properties

this would be doubly helpful since we're also using inverse(viewMatrix) in the vertex shader still